pkg/sql/sem/builtins/builtins.go

```diff
@@ -5022,6 +5022,32 @@ value if you rely on the HLC for accuracy.`,
 			Volatility: volatility.Stable,
 		},
 	),
+	"crdb_internal.descriptor_with_post_deserialization_changes": makeBuiltin(
+		tree.FunctionProperties{
+			Category:         builtinconstants.CategorySystemInfo,
+			DistsqlBlocklist: true,
+			Undocumented:     true,
+		},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"descriptor", types.Bytes}},
+			ReturnType: tree.FixedReturnType(types.Bytes),
+			Fn: func(ctx context.Context, evalCtx *eval.Context, args tree.Datums) (tree.Datum, error) {
+				s, ok := tree.AsDBytes(args[0])
+				if !ok {
+					return nil, errors.Newf("expected bytes value, got %T", args[0])
+				}
+				ret, err := evalCtx.CatalogBuiltins.DescriptorWithPostDeserializationChanges(
+					ctx, []byte(s), evalCtx.Txn.ReadTimestamp(),
```

Comment:
This choice of mvcc timestamp is confusing and non-obvious. I think it may one day cause problems and it will certainly mislead readers. This timestamp has no meaning here except that it's non-zero. At least tell somebody that. I'd rather that we buried the details about wanting to set the timestamp deeper than builtins. I also don't love that the descriptor coming out of this will have an arbitrary timestamp set on it.

Reply:
For better or for worse, descriptors with versions > 1 need to have a modification time set and that modification time is not persisted in the protobuf. We need to supply _something_. 

I think what I'll do is supply a sentinel value that is always valid and strip it out before re-encoding the descriptor the same way `MaybeIncrementVersion` would.

> I'd rather that we buried the details about wanting to set the timestamp deeper than builtins.

So do I.